From 9fca723f0023f45cb08794ce9b8e0a9f296fa049 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Tue, 11 Aug 2015 18:04:35 +0800
Subject: [PATCH 0443/1691] MLK-10879-03 cpufreq: imx: add cpufreq support for
 i.mx6ul.

commit 9fca723f0023f45cb08794ce9b8e0a9f296fa049 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Add cpufreq driver for i.MX6UL for DVFS. According to the design team, the ARM core can run
at setpoints 528MHz, 396MHz and 198MHz. This SOC adds a secondary_clk_sel(pll2/pll2_pfd2)
as the clock source of step_clk. In normal situation, just use the step_clk as the ARM
clk source, the pll1_sys is only used when change the cpu frequency.

Conflicts:
	arch/arm/mach-imx/mach-imx6ul.c
	drivers/cpufreq/imx6q-cpufreq.c

Signed-off-by: Bai Ping <b51503@freescale.com>
(cherry picked from commit f6528f7922c2ba1cacabf93eecb9301fb20d459e)
---
 arch/arm/mach-imx/mach-imx6ul.c |  6 +++++
 drivers/cpufreq/imx6q-cpufreq.c | 53 ++++++++++++++++++++++++++++++++++++-----
 2 files changed, 53 insertions(+), 6 deletions(-)

diff --git a/drivers/cpufreq/imx6q-cpufreq.c b/drivers/cpufreq/imx6q-cpufreq.c
index 81cf2c7..a86265d 100644
--- a/drivers/cpufreq/imx6q-cpufreq.c
+++ b/drivers/cpufreq/imx6q-cpufreq.c
@@ -30,6 +30,9 @@ static struct clk *pll1_sw_clk;
 static struct clk *pll1_sw_clk;
 static struct clk *step_clk;
 static struct clk *pll2_pfd2_396m_clk;
+static struct clk *pll1_bypass;
+static struct clk *pll1_bypass_src;
+static struct clk *pll1;
 
 /* clk used by i.MX6UL */
 static struct clk *pll2_bus_clk;

-- 
1.9.1

